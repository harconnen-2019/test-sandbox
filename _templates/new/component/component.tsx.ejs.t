---
to: <%= absPath %>/<%= componentName %>.tsx
---
import React from 'react';
import styles from './style.module.css';

type Props = {};

export const <%= componentName %>: React.FC<Props> = (props) => {
  return <div className={styles.container} data-testid="test" />;
};