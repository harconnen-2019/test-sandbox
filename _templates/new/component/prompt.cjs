module.exports = {
  prompt: ({ inquirer }) => {
    const questions = [
      {
        type: 'select',
        name: 'category',
        message: 'Определите слой?',
        choices: ['app', 'pages', 'widgets', 'features', 'entities', 'shared'],
      },
      {
        type: 'input',
        name: 'componentName',
        message: 'What is the component name?',
      },
      {
        type: 'input',
        name: 'dir',
        message: 'Where is the directory(Optional)',
      },
    ];
    return inquirer.prompt(questions).then((answers) => {
      const { category, componentName, dir } = answers;
      // eslint-disable-next-line sonarjs/no-nested-template-literals
      const path = `${category}/${dir ? `${dir}/` : ''}${componentName}`;
      const absPath = `src/${path}`;
      return {
        ...answers,
        path,
        absPath,
        category,
      };
    });
  },
};
