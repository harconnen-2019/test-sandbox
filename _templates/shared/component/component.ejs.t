---
to: src/shared/ui/<% if(locals.section) { -%><%= h.inflection.dasherize(section.toLowerCase()) %>/<%}-%><%= h.inflection.dasherize(name.toLowerCase()) %>/<%= h.inflection.dasherize(name.toLowerCase()) %>.tsx
---
<% ComponentName = h.inflection.camelize(name.split('-').join('_').split(' ').join('_')) -%>
import { ReactNode } from 'react';
<% if(locals.module) { -%>

import styles from './<%= h.inflection.dasherize(name.toLowerCase()) %>.module.css';
<%}-%>

type <%= ComponentName %>Props = {
  children: ReactNode;
};

function <%= ComponentName %>({ children }: <%= ComponentName %>Props) {
  return <div<% if(locals.module) { -%> className={styles.container}<%}-%><% if(locals.test) { -%> data-testid="<%= ComponentName %>"<%}-%>>{children}</div>;
}

export { <%= ComponentName %> };
