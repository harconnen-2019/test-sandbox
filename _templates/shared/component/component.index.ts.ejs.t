---
to: src/shared/ui/<% if(locals.section) { -%><%= h.inflection.dasherize(section.toLowerCase()) %>/<%}-%><%= h.inflection.dasherize(name.toLowerCase()) %>/index.ts
---
<% ComponentName = h.inflection.camelize(name.split('-').join('_').split(' ').join('_')) -%>
export * from './<%= h.inflection.dasherize(name.toLowerCase()) %>';
