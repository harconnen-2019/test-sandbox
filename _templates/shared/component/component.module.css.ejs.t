---
to: <% if(locals.module) { -%>src/shared/ui/<% if(locals.section) { -%><%= h.inflection.dasherize(section.toLowerCase()) %>/<%}-%><%= h.inflection.dasherize(name.toLowerCase()) %>/<%= h.inflection.dasherize(name.toLowerCase()) %>.module.css<% } else { null }-%>
---
.container {
  width: 100%;
}