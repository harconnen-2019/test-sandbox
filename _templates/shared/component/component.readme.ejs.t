---
to: <% if(locals.readme) { -%>src/shared/ui/<% if(locals.section) { -%><%= h.inflection.dasherize(section.toLowerCase()) %>/<%}-%><%= h.inflection.dasherize(name.toLowerCase()) %>/README.md<% } else { null }-%>
---
# <%= h.capitalize(Name) %>


## Props
| Свойство | Тип | Опционально | По умолчанию | Описание |
| ---------|-----|-------------|--------------|----------|


## Example
