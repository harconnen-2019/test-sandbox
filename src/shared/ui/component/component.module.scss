.container {
  width: 100%;
  color: aqua;

  &:hover {
    color: #fff;
  }

  p {
    color: red;
  }

  @media (orientation: landscape) {
    width: 500px;
  }
}
