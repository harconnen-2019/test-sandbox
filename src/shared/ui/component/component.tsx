import { ReactNode } from 'react';

import styles from './component.module.scss';

type ComponentProps = {
  children: ReactNode;
};

/**
 * [ ] не выбран
 * [x] выбран
 * @param props ComponentProps
 * @param props.children ReactNode
 * @returns Компонент с параметрами 1
 */
function Component({ children }: ComponentProps) {
  return (
    <div className={styles.container} data-testid="Component">
      {children}
    </div>
  );
}

export { Component };
