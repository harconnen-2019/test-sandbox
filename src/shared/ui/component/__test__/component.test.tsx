import { render, screen } from '@testing-library/react';

import { Component } from '../component';

describe('/shared/ui/component', () => {
  it('Название 1', () => {
    render(<Component>test</Component>);
    const element = screen.getByTestId(/Component/i);
    expect(element).toBeInTheDocument();
    expect(element).toMatchSnapshot();
  });
});
