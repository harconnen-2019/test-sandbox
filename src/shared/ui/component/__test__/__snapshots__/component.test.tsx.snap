// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/shared/ui/component Название 1 1`] = `
<div
  class="container"
  data-testid="Component"
>
  test
</div>
`;
